transport/transport-runtime/src/main/java/com/google/android/datatransport/runtime/scheduling/persistence/ClientAnalyticsStore.java

```diff
@@ -0,0 +1,47 @@
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.android.datatransport.runtime.scheduling.persistence;
+
+import com.google.android.datatransport.runtime.firebase.transport.ClientMetrics;
+import com.google.android.datatransport.runtime.firebase.transport.LogEventDropped.Reason;
+
+/**
+ * Persistent Layer
+ *
+ * <p>Responsible for storing, updating, and retrieving client analytics data.
+ */
+public interface ClientAnalyticsStore {
+  /**
+   * Record log event dropped.
+   *
+   * @param eventsDroppedCount number of events dropped
+   * @param reason reason why events are dropped
+   * @param logSource log source of the dropped events
+   */
+  void recordLogEventDropped(long eventsDroppedCount, Reason reason, String logSource);
```

Comment:
Curious to know when this method will be called, i.e. will `eventsDroppedCount` ever be greater than `1`?

Reply:
It will check if there is an existing row for that `reason` and `logSource`, if there is, it will increment the count value, if not, it will insert a new row.

Reply:
I get that, but my question is: will we ever pass anything other than `1` as input to this method?

Reply:
yes, we will. eg. if we trying to delete events which is older than 1 week, we would delete multiple events at that time.